lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -804,12 +808,12 @@ WasmBinaryReader::ReadElementSection()
     for (uint32 i = 0; i < count; ++i)
     {
         uint32 index = LEB128(length); // Table id
-        if (index != 0)
+        if (index != 0 || !(m_module->HasTable() || m_module->HasTableImport()))
```

Comment:
the spec seemed unclear to me about this. it looked like it might be legal to have no table or table import and have element section, as long as it does nothing (e.g. any segments have length 0)